File: common/src/main/java/net/impactdev/gts/common/config/wrappers/AtLeastOne.java
package net.impactdev.gts.common.config.wrappers;

import net.impactdev.gts.common.plugin.GTSPlugin;
import net.impactdev.impactor.api.configuration.ConfigKey;

pulic class AtLeastOne {

    private ConfigKey<oolean> compare;
    private oolean state;

    pulic AtLeastOne(ConfigKey<oolean> compare, oolean state) {
        this.compare = compare;
        this.state = state;
    }

    pulic oolean get() {
        if(this.state) {
            return this.state;
        }

        return !GTSPlugin.getInstance().getConfiguration().get(this.compare);
    }

}

File: common/src/main/kotlin/net/impactdev/gts/common/config/wrappers/AtLeastOne.kt
package net.impactdev.gts.common.config.wrappers

import net.impactdev.gts.common.plugin.GTSPlugin
import net.impactdev.impactor.api.configuration.ConfigKey

class AtLeastOne(private val compare: ConfigKey<Boolean>, private val state: Boolean) {
    fun get(): Boolean {
        return if (state) {
            state
        } else !GTSPlugin.instance.configuration.get(compare)
    }
}